packages/server/graphql/mutations/addFeatureFlag.ts

```diff
@@ -31,14 +34,27 @@ export default {
   },
   async resolve(
     _source: unknown,
-    {emails, domain, flag}: {emails: string[] | null; domain: string | null; flag: string},
+    {
+      emails,
+      domain,
+      flag
+    }: {emails: string[] | null; domain: string | null; flag: UserFeatureFlagEnum},
     {authToken, dataLoader}: GQLContext
   ) {
     const operationId = dataLoader.share()
     const subOptions = {operationId}
 
     // AUTH
-    requireSU(authToken)
+    const viewerId = getUserId(authToken)
+    const viewer = await getUserById(viewerId)
+    const isAddingFlagToViewer = !emails?.length && !domain
+    if (!isAddingFlagToViewer) {
+      requireSU(authToken)
```

Comment:
this is going to throw an error instead of returning an object with an error property.
any thrown errors have their message replaced with the string "Server Error" before being sent to the client.
so, for example, if we have a bug in our frontend code that adds a value to the domain prop, the user will just get a cryptic message

Reply:
I see! So is the reason that we throw an error from `requireSU` elsewhere in the codebase because we want to keep the error message ambiguous for potential attackers? Whereas it's a bad idea to use it here because I'm conditionally calling `requireSU` based on the mutation's arguments? So if the frontend code accidentally adds a domain prop, we want to send more helpful feedback because it's more likely to be a bug from a dev rather than an attack.

Reply:
it's really just an old pattern from back when we didn't have an `error` field as part of the response payload. doing it on internal calls (e.g. a stripe webhook calls our graphql server) is fine because it'll still get logged to sentry, but it'll never make it down to the client